Sources/PostgresNIO/Connection/PostgresDatabase+PreparedQuery.swift

```diff
@@ -0,0 +1,161 @@
+import Foundation
+
+extension PostgresDatabase {
+    public func prepare(query: String) -> EventLoopFuture<PreparedQuery> {
+        let name = "nio-postgres-\(UUID().uuidString)"
+        let prepare = PrepareQueryRequest(query, as: name)
+        return self.send(prepare, logger: self.logger).map { () -> (PreparedQuery) in
+            let prepared = PreparedQuery(database: self, name: name, rowDescription: prepare.rowLookupTable!)
+            return prepared
+        }
+    }
+
+    public func prepare(query: String, handler: @escaping (PreparedQuery) -> EventLoopFuture<[[PostgresRow]]>) -> EventLoopFuture<[[PostgresRow]]> {
+        prepare(query: query)
+        .flatMap { preparedQuery in
+            handler(preparedQuery)
+            .flatMap { results in
+                preparedQuery.deallocate().map { results }
+            }
+        }
+    }
+}
+
+
+public struct PreparedQuery {
+    let database: PostgresDatabase
+    let name: String
+    let rowLookupTable: PostgresRow.LookupTable
+
+    init(database: PostgresDatabase, name: String, rowDescription: PostgresRow.LookupTable) {
+        self.database = database
+        self.name = name
+        self.rowLookupTable = rowDescription
+    }
+
+    public func execute(_ binds: [PostgresData] = []) -> EventLoopFuture<[PostgresRow]> {
+        var rows: [PostgresRow] = []
+        return self.execute(binds) { rows.append($0) }.map { rows }
+    }
+
+    public func execute(_ binds: [PostgresData] = [], _ onRow: @escaping (PostgresRow) throws -> ()) -> EventLoopFuture<Void> {
+        let handler = ExecutePreparedQuery(query: self, binds: binds, onRow: onRow)
+        return database.send(handler, logger: database.logger)
+    }
+
+    public func deallocate() -> EventLoopFuture<Void> {
+        return database.query("DEALLOCATE \"\(name)\"").map { _ in }
```

Comment:
I think this should be done using the special close message instead of a separate query. This can be added similar to other `PostgresMessage+...` 

From: https://www.postgresql.org/docs/9.3/protocol-flow.html

> The Close message closes an existing prepared statement or portal and releases resources. It is not an error to issue Close against a nonexistent statement or portal name. The response is normally CloseComplete, but could be ErrorResponse if some difficulty is encountered while releasing resources. Note that closing a prepared statement implicitly closes any open portals that were constructed from that statement.

From: https://www.postgresql.org/docs/9.3/protocol-message-formats.html

```
Close (F)
Byte1('C')
Identifies the message as a Close command.

Int32
Length of message contents in bytes, including self.

Byte1
'S' to close a prepared statement; or 'P' to close a portal.

String
The name of the prepared statement or portal to close (an empty string selects the unnamed prepared statement or portal).

CloseComplete (B)
Byte1('3')
Identifies the message as a Close-complete indicator.

Int32(4)
Length of message contents in bytes, including self.
````

Reply:
Yeah, good point, fixed in the last commit.